tests/Makefile

```diff
@@ -88,13 +82,23 @@ APPS =	test_suite_aes.ecb$(EXEXT)	test_suite_aes.cbc$(EXEXT)	\
 
 .SILENT:
 
-.PHONY: all check test clean
+.PHONY: all all_c check clean dynamic_libraries test
 
 all: $(APPS)
 all_c: $(APPS:$(EXEXT)=.c)
 
-$(DEP):
+ifndef SHARED
+STATIC_LIBRARIES = ../library/libmbedcrypto.a ../library/libmbedx509.a ../library/libmbedtls.a
+DYNAMIC_LIBRARIES =
+$(STATIC_LIBRARIES):
+	$(MAKE) -C ../library
+else
+STATIC_LIBRARIES =
+DYNAMIC_LIBRARIES = ../library/libmbedcrypto.$(DLEXT) ../library/libmbedx509.$(DLEXT) ../library/libmbedtls.$(DLEXT)
+$(DYNAMIC_LIBRARIES):
 	$(MAKE) -C ../library
+endif
+dynamic_libraries: $(DYNAMIC_LIBRARIES)
```

Comment:
I there a reason why you have a target `dynamic_libraries` while you use `$(STATIC_LIBRARIES)` directly in dependencies lists? (Not that I think there's anything wrong with it, I just don't understand the reason for the asymmetry so perhaps I'm missing something.)